citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -979,6 +972,27 @@ protected void doPostJsonResponse(ParameterParser params,
 		return results;
 	}
 
+	protected Map<String, Object> addSection(ParameterParser params, SessionState state) {
+		String message;
+		Map<String, Object> results = new HashMap<String, Object>();
+		try {
+			String addSectionHTML = params.getString("addSectionHTML");
```

Comment:
You need to add some filtering for bad HTML so that you can't put evil script tags in.